lib/slack/real_time/concurrency/celluloid.rb

```diff
@@ -37,12 +37,11 @@ def run_loop
           rescue EOFError, Errno::ECONNRESET, Errno::EPIPE => e
             logger.debug("#{self.class}##{__method__}") { e }
             driver.emit(:close, WebSocket::Driver::CloseEvent.new(1001, 'server closed connection')) unless @closing
-          ensure
-            begin
-              current_actor.terminate if current_actor.alive?
-            rescue StandardError
-              nil
-            end
+          end
+
+          def disconnect!
+            super
+            current_actor.terminate if current_actor.alive?
```

Comment:
This line of code would have never worked because you cannot terminate your current thread (i.e. `current_actor`). It consistently throws an error.
```Ruby
ThreadError:
       Target thread must not be current thread
```